utopia-remix/app/handlers/validators.ts

```diff
@@ -0,0 +1,34 @@
+import { AccessValidator, ensure, getUser } from '../util/api.server'
+import { UserProjectPermission } from '../types'
+import { Params } from '@remix-run/react'
+import { Status } from '../util/statusCodes'
+import { hasUserProjectPermission } from '../services/permissionsService.server'
+import { getProjectOwnerById } from '../models/project.server'
+
+export function validateProjectAccess(
+  permission: UserProjectPermission,
+  {
+    errorMessage,
+    status,
+  }: {
+    errorMessage?: string
+    status?: number
+  } = {},
+): AccessValidator {
+  return async function (req: Request, params: Params<string>) {
+    const { id: projectId } = params
```

Comment:
the problem here is that this is requiring the route to be defined with `$projectId` in the route name. for example, all `internal.project.` routes use instead `$id`. To make things simpler either we rename all the routes to use `$projectId` (but it means that we need to remember to do it like this _every time_), or we instead pass the right param key when invoking the validator

Reply:
This is doing the opposite (asserting that the route is named `$id` and then renaming the param to `projectId`), but the point about the naming being implicitly required is valid here, because there will definitely be routes (e.g. `v1.github.save.$projectId`) where it makes more sense to use `$projectId` rather than `$id`.

Reply:
yeah, sorry, flipped the naming but the argument stands

Reply:
isn't it actually requiring `$id` in the project name and just renaming it to `projectId`?

Reply:
yeah, see @Rheeseyb comment

Reply:
added an option to pass additional config to the validator

Reply:
(writing here as well for completion - changed to a function that gets the project id)